pkg/operator/api/userconfig/apis.go

```diff
@@ -118,24 +119,27 @@ func (apis APIs) Validate() error {
 func (api *API) Validate() error {
 	if yaml.StartsWithEscapedAtSymbol(api.Model) {
 		api.ModelFormat = TensorFlowModelFormat
-	} else {
-		if !aws.IsValidS3Path(api.Model) {
-			return errors.Wrap(ErrorInvalidS3PathOrResourceReference(api.Model), Identify(api), ModelKey)
+		if err := api.Compute.Validate(); err != nil {
+			return errors.Wrap(err, Identify(api), ComputeKey)
 		}
 
-		if api.ModelFormat == UnknownModelFormat {
-			if strings.HasSuffix(api.Model, ".onnx") {
-				api.ModelFormat = ONNXModelFormat
-			} else if strings.HasSuffix(api.Model, ".zip") {
-				api.ModelFormat = TensorFlowModelFormat
-			} else {
-				return errors.Wrap(ErrorUnableToInferModelFormat(), Identify(api))
-			}
-		}
+		return nil
+	}
 
+	if !aws.IsValidS3Path(api.Model) {
+		return errors.Wrap(ErrorInvalidS3PathOrResourceReference(api.Model), Identify(api), ModelKey)
+	}
+
+	switch {
+	case strings.HasSuffix(api.Model, ".onnx"):
+		api.ModelFormat = ONNXModelFormat
 		if ok, err := aws.IsS3PathFileExternal(api.Model); err != nil || !ok {
 			return errors.Wrap(ErrorExternalNotFound(api.Model), Identify(api), ModelKey)
 		}
+	case models.IsValidS3Directory(api.Model):
+		api.ModelFormat = TensorFlowModelFormat
+	default:
+		return errors.Wrap(ErrorUnableToInferModelFormat(), Identify(api))
```

Comment:
The message in `ErrorUnableToInferModelFormat` needs to be updated. Can you describe something like you wrote in you comment in `models/models.go`? I can see users hitting this, so we should be as clear as possible in this error message, it's ok if it ends up being a bit long.